syft/pkg/cataloger/swift/parse_package_resolved.go

```diff
@@ -0,0 +1,134 @@
+package swift
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io"
+
+	"github.com/anchore/syft/syft/artifact"
+	"github.com/anchore/syft/syft/file"
+	"github.com/anchore/syft/syft/pkg"
+	"github.com/anchore/syft/syft/pkg/cataloger/generic"
+)
+
+var _ generic.Parser = parsePackageResolved
+
+// swift package manager has two versions (1 and 2) of the resolved files, the types below describes the serialization strategies for each version
+// with its suffix indicating which version its specific to.
+
+type packageResolvedV1 struct {
+	PackageObject packageObjectV1 `json:"object"`
+	Version       int             `json:"version"`
+}
+
+type packageObjectV1 struct {
+	Pins []packagePinsV1
+}
+
+type packagePinsV1 struct {
+	Name          string       `json:"package"`
+	RepositoryURL string       `json:"repositoryURL"`
+	State         packageState `json:"state"`
+}
+
+type packageResolvedV2 struct {
+	Pins []packagePinsV2
+}
+
+type packagePinsV2 struct {
+	Identity string       `json:"identity"`
+	Kind     string       `json:"kind"`
+	Location string       `json:"location"`
+	State    packageState `json:"state"`
+}
+
+type packagePin struct {
+	Identity string
+	Location string
+	Revision string
+	Version  string
+}
+
+type packageState struct {
+	Revision string `json:"revision"`
+	Version  string `json:"version"`
+}
+
+// parsePackageResolved is a parser for the contents of a Package.resolved file, which is generated by Xcode after it's resolved Swift Package Manger packages.
```

Comment:
Are there any relationships we should consider between `Packages` that are resolved from the `./Package.resolved` vs `./.builds/checkout` or other locations that a `.resolved` file might be found?

Reply:
The relationship to be considered could be the target specific ones, if a resolved file is in only a certain target directory, it might indicate it's only required for that specific target.

I'm actually a bit uncertain as far as `./.builds/` go (once again no Swift Package Manager expert), I don't remember seeing that in my testing, but I do know that the location of the resolved file does depend on how you resolved, so it differs from CLI and Xcode, when it comes to resolving packages. So could be why I did not see that file in my testing?